Add link and better details on the context parameter. (#6309)
cakephp/docs: en/core-libraries/validation.rst

used. In general, it is recommended that you do not use ``notEmpty()`` and use more
specific validators instead: ``notEmptyString()``, ``notEmptyArray()``, ``notEmptyFile()``, ``notEmptyDate()``, ``notEmptyTime()``, ``notEmptyDateTime()``.

The ``allowEmpty*`` methods support a ``when`` parameter that allows you to control
when a field can or cannot be empty:

* ``false`` The field is not allowed to be empty.
* ``create`` The field can be empty when validating a **create**
  operation.
* ``update`` The field can be empty when validating an **update**
  operation.
* A callback that returns ``true`` or ``false`` to indicate whether a field is

>   required or not. See the :ref:`conditional-validation` section for examples on

Reviewer: ```suggestion
  allowed to be empty. See the :ref:`conditional-validation` section for examples on
```
